portalnet/src/overlay.rs

```diff
@@ -441,6 +441,43 @@ where
         }
     }
 
+    pub async fn lookup_node(&self, target: NodeId) -> Vec<Enr> {
+        if target == self.local_enr().node_id() {
+            return vec![self.local_enr()];
+        }
+        let connected_peer = self
+            .kbuckets
+            .write()
+            .iter()
+            .filter(|entry| entry.status.is_connected())
+            .map(|entry| *entry.node.key.preimage())
+            .find(|node_id| node_id == &target);
+        if let Some(entry) = connected_peer {
+            let enr = self.discovery.find_enr(&entry).expect("ENR should exist");
```

Comment:
Panicking here seems a little extreme. Sure, it shouldn't happen, but do we really want to crash trin if we have an entry in our kbuckets that the discv5 lib forgot about?

I'd say an `error!()` log and an empty `vec` would be sufficient.